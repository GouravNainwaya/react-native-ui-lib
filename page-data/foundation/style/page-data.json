{"componentChunkName":"component---src-templates-markdown-template-js","path":"/foundation/style","result":{"data":{"markdownRemark":{"html":"<p>The base foundation of each UI component is its style.\nWe use basic style presets to define the rules and the style guide we follow.</p>\n<p>Our presets include: <strong>Colors</strong>, <strong>Typography</strong>, <strong>Spacings</strong>, <strong>Shadows</strong>, <strong>Border Radius</strong> and more..</p>\n<p>The UILib already comes with a set of predefined constants and <a href=\"https://github.com/wix/react-native-ui-lib/tree/master/src/style\">presets</a>.</p>\n<p>You can easily use it anywhere in your code (as you would use any other constant value), or as a component modifier.</p>\n<p>It's also very easy to define your own presets..</p>\n<pre><code class=\"language-jsx\">import {Typography, Colors, Spacings} from 'react-native-ui-lib';\n\nColors.loadColors({\n  pink: '#FF69B4',\n  gold: '#FFD700',\n});\n\nTypography.loadTypographies({\n  h1: {fontSize: 58, fontWeight: '300', lineHeight: 80},\n  h2: {fontSize: 46, fontWeight: '300', lineHeight: 64},\n});\n\nSpacings.loadSpacings({\n  page: isSmallScreen ? 16 : 20\n});\n</code></pre>\n<p>For example, the following line</p>\n<pre><code class=\"language-jsx\">&#x3C;Text h1 pink>Hello World&#x3C;/Text>\n</code></pre>\n<p>Will generate this text\n<img src=\"https://cloud.githubusercontent.com/assets/1780255/24792314/296b7ebc-1b86-11e7-8580-9252d1ddf5d9.png\" width=\"250\"/></p>\n<p>It will use the <em>h1</em> preset for typography and the <em>pink</em> color value we set to style the Text element.</p>","frontmatter":{"path":"/foundation/style","title":"Style"}}},"pageContext":{}},"staticQueryHashes":["1243665866","3675868689"]}