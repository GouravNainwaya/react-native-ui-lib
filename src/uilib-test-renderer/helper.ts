import _ from 'lodash';

interface Style {
  style: any;
}

interface Props {
  props: Style;
}

const findStyle = <T>(key: string, component: Props): T => {
  const flat = _.flatMap(component.props.style) as Array<any | undefined>;
  const color = _.find(flat, key)[key];
  return color;
};

export {findStyle};
