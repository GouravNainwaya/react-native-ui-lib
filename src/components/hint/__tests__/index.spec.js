import React from 'react';
import {Hint} from 'react-native-ui-lib';
import {Colors} from '../../../style';
import {render, findStyle} from '../../../uilib-test-renderer';
// import HintsScreen from '../../../../demo/src/screens/componentScreens/HintsScreen';

const HintTestComponent = ({showHint}) => {
  return (
    <Hint
      visible={showHint}
      message={'Hint message to hint things'}
      position={Hint.positions.TOP}
      useSideTip
      key={'1'}
      targetFrame={{x: 1, y: 1, height: 1, width: 1}}
      onBackgroundPress={() => {}}
      color={Colors.white}
      removePaddings
      enableShadow
      testID={'Hint'}
    />
  );
};

describe('Hint Screen component test', () => {
  it('Test Hint component background color', async () => {
    const hintTestId = 'Hint.message';

    const expectedColor = Colors.primary;
    const element = <HintTestComponent useTargetFrame showHint/>;

    const {getByTestId, toJSON} = render(element);

    const wrapper = getByTestId('Hint');
    expect(wrapper).toBeTruthy();

    const hint = getByTestId(hintTestId);
    const color = findStyle('backgroundColor', hint);

    expect(color).toBe(expectedColor);
    expect(hint).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('Test Hint modal is not visible when showHint is false', async () => {
    const element = <HintTestComponent useTargetFrame showHint={false}/>;

    const {toJSON, queryByTestId} = render(element);

    const modal = queryByTestId('Hint.modal');

    expect(modal).toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });

  it('Test Hint modal is visible when showHint is true', async () => {
    const element = <HintTestComponent useTargetFrame showHint/>;

    const {getByTestId, queryAllByTestId, toJSON} = render(element);
    const hint = getByTestId('Hint');
    expect(hint).toBeTruthy();
  
    expect(queryAllByTestId('Hint.modal')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
