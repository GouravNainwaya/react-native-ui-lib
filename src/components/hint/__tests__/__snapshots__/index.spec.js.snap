// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint Screen component test Test Hint component background color 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  modifiers={
    Object {
      "alignments": Object {},
      "backgroundColor": undefined,
      "borderRadius": undefined,
      "color": "transparent",
      "flexStyle": undefined,
      "margins": Object {},
      "paddings": Object {},
      "positionStyle": undefined,
      "typography": undefined,
    }
  }
  onBackgroundPress={[Function]}
  onRequestClose={[Function]}
  testID="Hint.modal"
  transparent={true}
  visible={true}
>
  <View
    blurType="light"
    style={
      Array [
        undefined,
        undefined,
        undefined,
        undefined,
        Object {},
        Object {},
        Object {},
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          undefined,
          undefined,
          undefined,
          Object {},
          Object {},
          Object {},
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": undefined,
            },
          ],
        ]
      }
      testID="Hint.modal"
      useSafeArea={false}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            undefined,
            undefined,
            undefined,
            undefined,
            Object {},
            Object {},
            Object {},
            Object {
              "flex": 1,
            },
          ]
        }
      />
    </View>
    <View
      collapsable={true}
      color="#FFFFFF"
      enableShadow={true}
      message="Hint message to hint things"
      onBackgroundPress={[Function]}
      position="top"
      removePaddings={true}
      style={
        Array [
          undefined,
          undefined,
          undefined,
          undefined,
          Object {},
          Object {},
          Object {},
          Array [
            Object {
              "position": "absolute",
            },
            undefined,
            Object {
              "left": 1,
              "top": 1,
            },
          ],
        ]
      }
      targetFrame={
        Object {
          "height": 1,
          "width": 1,
          "x": 1,
          "y": 1,
        }
      }
      useSideTip={true}
      visible={true}
    >
      <View
        animated={true}
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "flex-start",
            "bottom": 0,
            "left": -1,
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingLeft": 1,
            "paddingVertical": 16,
            "position": "absolute",
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
            "width": 750,
          }
        }
        testID="Hint"
      >
        <View
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            centerV={true}
            row={true}
            style={
              Array [
                undefined,
                undefined,
                undefined,
                undefined,
                Object {},
                Object {},
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Array [
                  Object {
                    "backgroundColor": "#5847FF",
                    "borderRadius": 20,
                    "maxWidth": 400,
                  },
                  false,
                  Object {
                    "elevation": 4,
                    "shadowColor": "#79838A",
                    "shadowOffset": Object {
                      "height": 5,
                      "width": 0,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 12,
                  },
                  Object {
                    "backgroundColor": "#FFFFFF",
                  },
                  false,
                ],
              ]
            }
            testID="Hint.message"
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "textAlign": "left",
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {},
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Hint message to hint things
            </Text>
          </View>
        </View>
        <Image
          accessibilityRole="image"
          accessible={false}
          assetGroup="icons"
          onError={[Function]}
          source={
            Object {
              "testUri": "../../../src/components/hint/assets/hintTipSide.png",
            }
          }
          style={
            Array [
              Object {
                "tintColor": "#FFFFFF",
              },
              undefined,
              undefined,
              undefined,
              undefined,
              Object {},
              Array [
                Object {
                  "position": "absolute",
                },
                Object {
                  "bottom": 9,
                  "left": 1,
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleY": -1,
                    },
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
              ],
            ]
          }
        />
      </View>
    </View>
  </View>
</Modal>
`;

exports[`Hint Screen component test Test Hint modal is not visible when showHint is false 1`] = `null`;

exports[`Hint Screen component test Test Hint modal is visible when showHint is true 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  modifiers={
    Object {
      "alignments": Object {},
      "backgroundColor": undefined,
      "borderRadius": undefined,
      "color": "transparent",
      "flexStyle": undefined,
      "margins": Object {},
      "paddings": Object {},
      "positionStyle": undefined,
      "typography": undefined,
    }
  }
  onBackgroundPress={[Function]}
  onRequestClose={[Function]}
  testID="Hint.modal"
  transparent={true}
  visible={true}
>
  <View
    blurType="light"
    style={
      Array [
        undefined,
        undefined,
        undefined,
        undefined,
        Object {},
        Object {},
        Object {},
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          undefined,
          undefined,
          undefined,
          Object {},
          Object {},
          Object {},
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": undefined,
            },
          ],
        ]
      }
      testID="Hint.modal"
      useSafeArea={false}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            undefined,
            undefined,
            undefined,
            undefined,
            Object {},
            Object {},
            Object {},
            Object {
              "flex": 1,
            },
          ]
        }
      />
    </View>
    <View
      collapsable={true}
      color="#FFFFFF"
      enableShadow={true}
      message="Hint message to hint things"
      onBackgroundPress={[Function]}
      position="top"
      removePaddings={true}
      style={
        Array [
          undefined,
          undefined,
          undefined,
          undefined,
          Object {},
          Object {},
          Object {},
          Array [
            Object {
              "position": "absolute",
            },
            undefined,
            Object {
              "left": 1,
              "top": 1,
            },
          ],
        ]
      }
      targetFrame={
        Object {
          "height": 1,
          "width": 1,
          "x": 1,
          "y": 1,
        }
      }
      useSideTip={true}
      visible={true}
    >
      <View
        animated={true}
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "flex-start",
            "bottom": 0,
            "left": -1,
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingLeft": 1,
            "paddingVertical": 16,
            "position": "absolute",
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
            "width": 750,
          }
        }
        testID="Hint"
      >
        <View
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            centerV={true}
            row={true}
            style={
              Array [
                undefined,
                undefined,
                undefined,
                undefined,
                Object {},
                Object {},
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Array [
                  Object {
                    "backgroundColor": "#5847FF",
                    "borderRadius": 20,
                    "maxWidth": 400,
                  },
                  false,
                  Object {
                    "elevation": 4,
                    "shadowColor": "#79838A",
                    "shadowOffset": Object {
                      "height": 5,
                      "width": 0,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 12,
                  },
                  Object {
                    "backgroundColor": "#FFFFFF",
                  },
                  false,
                ],
              ]
            }
            testID="Hint.message"
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "textAlign": "left",
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {},
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Hint message to hint things
            </Text>
          </View>
        </View>
        <Image
          accessibilityRole="image"
          accessible={false}
          assetGroup="icons"
          onError={[Function]}
          source={
            Object {
              "testUri": "../../../src/components/hint/assets/hintTipSide.png",
            }
          }
          style={
            Array [
              Object {
                "tintColor": "#FFFFFF",
              },
              undefined,
              undefined,
              undefined,
              undefined,
              Object {},
              Array [
                Object {
                  "position": "absolute",
                },
                Object {
                  "bottom": 9,
                  "left": 1,
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleY": -1,
                    },
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
              ],
            ]
          }
        />
      </View>
    </View>
  </View>
</Modal>
`;
